backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/workpackage/ModularContractLogHandler.java

```diff
@@ -123,18 +122,8 @@ private void handleLogs(
 			return;
 		}
 
-		final ProductId productId = handler.getProductId(request).orElse(null);
-		if (productId == null)
-		{
-			Loggables.withLogger(logger, Level.DEBUG)
-					.addLog("No Product found for contractId: {}! no logs will be created!", request.getContractId());
-
-			return;
-		}
-
-		final ModuleConfig moduleConfig = settings.getModuleConfig(handler.getComputingMethod().getComputingMethodType(), productId)
-				.orElse(null);
-		if (moduleConfig == null)
+		final List<ModuleConfig> moduleConfigs = settings.getModuleConfigs(handler.getComputingMethod().getComputingMethodType());
```

Comment:
consider introducing handler.getComputingMethodType().

see https://www.baeldung.com/java-demeter-law

Reply:
thx, It's also used here `de.metas.contracts.modular.workpackage.ModularContractLogHandlerRegistry` for the applies, but running out of time to get it integrated